### PR TITLE
EXSWHTEC-5 - Implement additional test for hipSetDevice

### PR DESCRIPTION
**Pull request by mirza-halilcevic**
- Test functionality in multithreaded environment.

Depends-on: #2887

### CODE REVIEW

File: tests/catch/unit/device/hipSetGetDevice.cc [hunk 1]
```diff
@@ -83,6 +87,46 @@ TEST_CASE("Unit_hipGetSetDevice_MultiThreaded") {
...
 }
 
+TEST_CASE("Unit_hipSetGetDevice_Positive_Threaded_Basic") {
+  class HipSetGetDeviceThreadedTest : public ThreadedZigZagTest<HipSetGetDeviceThreadedTest> {
+   public:
+    void TestPart1() { HIP_CHECK(hipSetDevice(0)); }
+
+    void TestPart2() {
+      HIP_CHECK_THREAD(hipSetDevice(1));
+      HIP_CHECK_THREAD(hipMalloc(&ptr, 2 * 1024 * 1024));
```
Comment (rorake) [inline tests/catch/unit/device/hipSetGetDevice.cc:96]: hipFree the allocated memory in ptr

File: tests/catch/unit/device/hipSetGetDevice.cc [hunk 2]
```diff
@@ -83,6 +87,46 @@ TEST_CASE("Unit_hipGetSetDevice_MultiThreaded") {
...
+      HIP_CHECK_THREAD(hipGetDevice(&device));
+      REQUIRE_THREAD(device == 0);
+      device = -1;
+      // To check if set device worked properly, outside of hipGetDevice
+      HIP_CHECK_THREAD(hipPointerGetAttribute(&device, HIP_POINTER_ATTRIBUTE_DEVICE_ORDINAL,
+                                              reinterpret_cast<hipDeviceptr_t>(ptr)));
+      REQUIRE_THREAD(device == 1);
+    }
+
+    void TestPart4() {
```
Comment (rorake) [inline tests/catch/unit/device/hipSetGetDevice.cc:108]: Remove the empty lines between TestPart1/2/3/4

File: tests/catch/unit/device/hipSetGetDevice.cc (hunk 1 above)
Comment (mirza-halilcevic) [reply tests/catch/unit/device/hipSetGetDevice.cc:96]: The suggested change is implemented. Please proceed with the review.

File: tests/catch/unit/device/hipSetGetDevice.cc (hunk 2 above)
Comment (mirza-halilcevic) [reply tests/catch/unit/device/hipSetGetDevice.cc:108]: The suggested change is implemented. Please proceed with the review.

**Review by rorake — APPROVED** [review]

**Review by chrispaquot — APPROVED** [review]

**Pull request merged**